### DISCUSSION TITLE
Is it possible to pass SRT through an nginx proxy?

### DISCUSSION THREAD

**Discussion by ieaster1** [Haivision/srt#2220] [Q&A] [answered]
This may be more of an nginx question but figured I would ask here in case any one else has attempted this.

I am able to see the traffic through the proxy and to the destination but it seems like the handshake is never being completed.  A specific note is I assumed only UDP would be necessary but is TCP also needed?  I wouldn't be surprised if I was misunderstanding the protocol used here.

At the moment I have a very basic config for the proxy_pass that I have tweaked here or there to see if it helps.

Here is the base config I've been attempting to use:
```
 server {
    listen 51935 udp;
    proxy_pass <srvr_ip>:51935;
    proxy_responses 1;
    proxy_timeout   5s;
}
```

**Comment by ieaster1** [comment, marked as answer]
In case anyone else comes across this, my suspicion was correct. Both TCP and UDP listen directives need to be set for traffic to be passed.  I'm not sure why this works but it resolved the issue for me.

Boiler plate config with the nginx stream module if anyone ever needs it:
```
 server {
    listen <port>;
    listen <port> udp;
    proxy_pass <srvr_ip>:<port>;
}
```

http://nginx.org/en/docs/stream/ngx_stream_core_module.html

**Comment by jeandube** [comment]
There is no TCP involved in SRT. But the handshake will most probably fails because the IP source address in the handshake packets as seen by the listener is not the original caller IP. I say this without knowing how nginx works.

**Reply by ieaster1** [reply]
I'm a little confused myself and I've probably just lucked out into a windfall.  I've edited my answer to prevent future confusion, but having both TCP and UDP listen directives _"resolves"_ my issue.  Mileage may vary.

**Reply by itseramin** [reply]
Any idea how to do it this in a Kubernetes cluster environment with the NGINX (ingress controller)?

**Reply by tab1293** [reply]
@itseramin Did you have any success getting SRT working with nginx-ingress?

**Reply by itseramin** [reply]
@tab1293 It turns out the limitation was on DigitalOcean's side, there was nothing wrong with NGINX, it is possible so expose UDP ports (https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/)... The actual limitation is with the Load Balancer: https://ideas.digitalocean.com/network/p/proxy-protocol-support-for-udp-http3

**Reply by tab1293** [reply]
@itseramin Thanks for your reply.

I'm wondering if you can provide some more guidance on how you got an SRT Caller -> Listener working through ingress-nginx. Can you share any kubernetes configs you may have?

It seems like the problem I'm running in to is the SRT handshake is failing because UDP traffic is not being correctly routed back to the SRT caller. More info in this issue: https://github.com/Haivision/srt/issues/3128
